### DISCUSSION TITLE
UG2 Seattle meeting

### DISCUSSION THREAD

**Discussion by jklymak** [c-proof/pyglider#70] [General]
Hi @callumrollo @hvdosser I think I'll try and present pyglider at the Seattle meeting.  Is this OK w/ both of you?  I will post an abstract here...

**Comment by callumrollo** [comment]
That's great! Would be good to gauge the interest from other UG2 members. Are there any improvements you want to implement before the meeting?

**Reply by jklymak** [reply]
The docs need to be fleshed out some more, and it should be at least pip installable, if not conda.  We should also work on a list of supported glider payloads.  I'm a little ignornant about what all is out there.  

Sorry, should have pinged @richardsc as well

**Reply by jklymak** [reply]
Oh, and the docs should be served somewhere.  Maybe readthedocs is easy enough....

**Reply by jklymak** [reply]
https://pyglider.readthedocs.io/en/latest/.  Seems OK.  need to add a link back to the GitHub repo...

**Comment by jklymak** [comment]
```
PyGlider is an open-source conversion toolchain that translates raw data from Slocum and SeaExplorer gliders and 
creates netCDF timeseries, profiles files, and depth-time gridded products.  The workflow creates raw netcdf files, 
which are as direct as possible translations of the glider raw data files. It then uses a YAML configuration file to 
create a CF-compliant netCDF timeseries file, extracted profiles, and a vertically gridded product.  We also supply 
rudimentary plotting routines.  The advantages of this pipeline over existing pipelines is that it is pure python, and 
hence does not rely on system-dependent executables and is all open source, including dependencies; it saves 
intermediate files so reprocessing is relatively straight forward; and most meta data is encoded in a mission-dependent 
YAML file that is easy to maintain and track in version control.  Disadvantages are that the initial parsing can be slow 
compared to executable code, and that the pipeline has not been tested and likely needs effort to make work with 
varied sensor payloads.  We welcome collaboration at https://github.com/c-proof/pyglider 
```

**Reply by jklymak** [reply]
ping @callumrollo @hvdosser Is this OK to submit with both your names on it?

**Reply by hvdosser** [reply]
It looks good! Happy for you to submit with my name on it.

**Reply by callumrollo** [reply]
Looks good to me! Happy to add my name.

If you want some operational stats, we've been testing the pipeline quite extensively at VOTO for 6 months now. We have pyglider in production processing NRT and delayed mode data from 8 gliders with various combinations of 7 sensors. 

Deployment yml https://github.com/voto-ocean-knowledge/deployment-yaml

Live missions and some total stats https://observations.voiceoftheocean.org/

**Reply by jklymak** [reply]
Thanks both.  Done!
